homeassistant/components/bluetooth/manager.py

```diff
@@ -225,15 +225,18 @@ def async_get_scanner_discovered_devices_and_advertisement_data_by_address(
         results: list[tuple[BaseHaScanner, BLEDevice, AdvertisementData]] = []
         for type_ in types_:
             for scanner in self._get_scanners_by_type(type_):
-                if device_advertisement_data := scanner.discovered_devices_and_advertisement_data.get(
+                if adv_data := scanner.discovered_devices_and_advertisement_data.get(
```

Comment:
```suggestion
                devices_and_adv_data = scanner.discovered_devices_and_advertisement_data
                if device_adv_data := devices_and_adv_data.get(address):
```

Reply:
This hard to read.  I'm going to refactor this a bit in #83733

Reply:
As its needed for a public in #83733 I renamed `async_get_scanner_discovered_devices_and_advertisement_data_by_address` -> `async_scanner_devices_by_address` and wrapped up the result in a dataclass so it should be a lot easier to read